deployment/cdk/opensearch-service-migration/lib/common-utilities.ts

```diff
@@ -6,6 +6,78 @@ import { IApplicationLoadBalancer } from "aws-cdk-lib/aws-elasticloadbalancingv2
 import { ICertificate } from "aws-cdk-lib/aws-certificatemanager";
 import { IStringParameter, StringParameter } from "aws-cdk-lib/aws-ssm";
 import * as forge from 'node-forge';
+import * as yargs from 'yargs';
+
+
+// parseAndMergeArgs, see @common-utilities.test.ts for an example of different cases
+export function parseAndMergeArgs(baseCommand: string, extraArgs?: string): string {
+    if (!extraArgs) {
+        return baseCommand;
+    }
+
+    // Extract command prefix
+    const commandPrefix = baseCommand.substring(0, baseCommand.indexOf('--')).trim();
+    const baseArgs = baseCommand.substring(baseCommand.indexOf('--'));
+
+    // Parse base command
+    const baseYargsConfig = {
+        parserConfiguration: {
+            'camel-case-expansion': false,
+            'boolean-negation': false,
+        }
+    };
+
+    const baseArgv = yargs(baseArgs)
+        .parserConfiguration(baseYargsConfig.parserConfiguration)
+        .parse();
+
+    // Parse extra args if provided
+    const extraYargsConfig = {
+        parserConfiguration: {
+            'camel-case-expansion': false,
+            'boolean-negation': true,
+        }
+    };
+
+    const extraArgv = extraArgs
+        ? yargs(extraArgs.split(' '))
+            .parserConfiguration(extraYargsConfig.parserConfiguration)
+            .parse()
+        : {};
+
+    // Merge arguments
+    const mergedArgv: { [key: string]: unknown } = { ...baseArgv };
+    for (const [key, value] of Object.entries(extraArgv)) {
+        if (key !== '_' && key !== '$0') {
+            if (!value &&
+                typeof value === 'boolean' &&
+                (
+                    typeof (baseArgv as any)[key] === 'boolean' ||
+                    (typeof (baseArgv as any)[`no-${key}`] != 'boolean' && typeof (baseArgv as any)[`no-${key}`])
+                )
+            ) {
+                delete mergedArgv[key];
+            } else {
+                mergedArgv[key] = value;
+            }
+        }
+    }
+
+    // Reconstruct command
+    const mergedArgs = Object.entries(mergedArgv)
+        .filter(([key]) => key !== '_' && key !== '$0')
+        .map(([key, value]) => {
+            if (typeof value === 'boolean') {
+                return value ? `--${key}` : `--no-${key}`;
+            }
+            return `--${key} ${value}`;
+        })
+        .join(' ');
```

Comment:
Is this list still in the order of the original command structure with the additional args in order at the end?  If not can we make it work that way to make diffing easier?

Reply:
It looks like this is the case, but I'm double checking